pkgs/by-name/gm/gmt/package.nix

```
@@ -0,0 +1,112 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  curl,
+  darwin,
+  fftwSinglePrec,
+  netcdf,
+  pcre,
+  gdal,
+  blas,
+  lapack,
+  glibc,
+  ghostscript,
+  dcw-gmt,
+  gshhg-gmt,
+  libxml2,
+}:
+/*
+  The onus is on the user to also install:
+   - ffmpeg for webm or mp4 output
+   - graphicsmagick for gif output
+*/
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gmt";
+  version = "6.5.0";
+  src = fetchFromGitHub {
+    owner = "GenericMappingTools";
+    repo = "gmt";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-KKIYhljCtk9t9CuvTLsSGvUkUwazWTm9ymBB3wLwSoI=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  env = {
+    NIX_LDFLAGS = "-lxml2 -L${lib.getLib (libxml2.override { enableHttp = true; })}/lib";
+  };
+
+  buildInputs =
+    [
+      curl
+      gdal
+      netcdf
+      pcre
+      dcw-gmt
+      gshhg-gmt
+    ]
+    ++ (
+      if stdenv.isDarwin then
+        with darwin.apple_sdk.frameworks;
+        [
+          Accelerate
+          CoreGraphics
+          CoreVideo
+        ]
+      else
+        [
+          glibc
+          fftwSinglePrec
+          blas
+          lapack
+        ]
+    );
+
+  propagatedBuildInputs = [ ghostscript ];
+
+  cmakeFlags =
+    [
+      "-DGMT_DOCDIR=share/doc/gmt"
+      "-DGMT_MANDIR=share/man"
+      "-DGMT_LIBDIR=lib"
+      "-DCOPY_GSHHG:BOOL=FALSE"
+      "-DGSHHG_ROOT=${gshhg-gmt.out}/share/gshhg-gmt"
+      "-DCOPY_DCW:BOOL=FALSE"
+      "-DDCW_ROOT=${dcw-gmt.out}/share/dcw-gmt"
+      "-DGDAL_ROOT=${gdal.out}"
+      "-DNETCDF_ROOT=${netcdf.out}"
+      "-DPCRE_ROOT=${pcre.out}"
+      "-DGMT_INSTALL_TRADITIONAL_FOLDERNAMES:BOOL=FALSE"
+      "-DGMT_ENABLE_OPENMP:BOOL=TRUE"
+      "-DGMT_INSTALL_MODULE_LINKS:BOOL=FALSE"
+      "-DLICENSE_RESTRICTED=LGPL" # "GPL" and "no" also valid
```

Use of `lib.cmakeFeature` and `lib.cmakeBool` would be helpful here:
```suggestion
      (lib.cmakeFeature "GMT_DOCDIR" "share/doc/gmt")
      (lib.cmakeFeature "GMT_MANDIR" "share/man")
      (lib.cmakeFeature "GMT_LIBDIR" "lib")
      (lib.cmakeBool "COPY_GSHHG" false)
      (lib.cmakeFeature "GSHHG_ROOT" "${gshhg-gmt.out}/share/gshhg-gmt")
      (lib.cmakeBool "COPY_DCW" false)
      (lib.cmakeFeature "DCW_ROOT" "${dcw-gmt.out}/share/dcw-gmt")
      (lib.cmakeFeature "GDAL_ROOT" "${gdal.out}")
      (lib.cmakeFeature "NETCDF_ROOT" "${netcdf.out}")
      (lib.cmakeFeature "PCRE_ROOT" "${pcre.out}")
      (lib.cmakeBool "GMT_INSTALL_TRADITIONAL_FOLDERNAMES" false)
      (lib.cmakeBool "GMT_ENABLE_OPENMP" true)
      (lib.cmakeBool "GMT_INSTALL_MODULE_LINKS" false)
      (lib.cmakeFeature "LICENSE_RESTRICTED" "LGPL")
```